--output-skb is cut short

Example: [pwru.log](https://github.com/user-attachments/files/19128055/pwru.log)

Started as `pwru --output-skb --filter-trace-tc 'port 80'`.

Version: [Release v1.0.10-pre](https://github.com/cilium/pwru/releases/tag/untagged-b7ad3f3c997fe992fe3e)

How can you set the buffer length of the char array before loading without the verifier complaining? I tried but it's not working. Thanks!!

```
TARGET_GOARCH=amd64 go generate
Generating for amd64
/home/hundi/pwru/bpf/kprobe_pwru.c:182:7: error: fields must have a constant size: 'variable length array in structure' extension will never be supported
        char str[PRINT_SKB_STR_SIZE];
             ^
/home/hundi/pwru/bpf/kprobe_pwru.c:186:7: error: fields must have a constant size: 'variable length array in structure' extension will never be supported
        char str[PRINT_SHINFO_STR_SIZE];
             ^
/home/hundi/pwru/bpf/kprobe_pwru.c:216:9: error: fields must have a constant size: 'variable length array in structure' extension will never be supported
        __type(value, u8[PERCPU_BIG_BUFF_SIZE]);
               ^
3 errors generated.
Error: compile: exit status 1
exit status 1
build.go:5: running "go": exit status 1
make: *** [Makefile:21: pwru] Error 1

```

--- a/bpf/kprobe_pwru.c
+++ b/bpf/kprobe_pwru.c
@@ -209,7 +209,7 @@ struct {
 struct {
 	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
 	__type(key, u32);
-	__type(value, u8[PERCPU_BIG_BUFF_SIZE]);
+	__uint(value_size, PERCPU_BIG_BUFF_SIZE);
 	__uint(max_entries, 1);
 } percpu_big_buff SEC(".maps");
 

--- a/internal/pwru/types.go
+++ b/internal/pwru/types.go
@@ -65,6 +65,8 @@ type Flags struct {
 	ReadyFile string
 
 	Backend string
+
+	SetPerCPUBuf uint32
 }
 
 func (f *Flags) SetFlags() {
@@ -110,6 +112,7 @@ func (f *Flags) SetFlags() {
 	flag.StringVar(&f.Backend, "backend", "",
 		fmt.Sprintf("Tracing backend('%s', '%s'). Will auto-detect if not specified.", BackendKprobe, BackendKprobeMulti))
 
+	flag.Uint32Var(&f.SetPerCPUBuf, "set-percpu-buf", 4096, "set the size of buffers to print skb data (used by --output-skb and --output-skb-shared-info)")
 	flag.Usage = func() {
 		fmt.Fprintf(os.Stderr, "Usage: %s [options] [pcap-filter]\n", os.Args[0])
 		fmt.Fprintf(os.Stderr, "    Available pcap-filter: see \"man 7 pcap-filter\"\n")

--- a/main.go
+++ b/main.go
@@ -195,6 +195,8 @@ func main() {
 		log.Fatalf("Failed to rewrite config: %v", err)
 	}
 
+	bpfSpec.Maps["percpu_big_buff"].ValueSize = flags.SetPerCPUBuf
+	
 	haveFexit := pwru.HaveBPFLinkTracing()
 	if (flags.FilterTraceTc || flags.FilterTraceXdp) && !haveFexit {
 		log.Fatalf("Current kernel does not support fentry/fexit to run with --filter-trace-tc/--filter-trace-xdp")

